Split the core web client into its own package.json

brianhelba opened the pull request:
This will allow the code within the core web client to be published and reused by other projects with different build systems.

zachmullen commented on docs/development.rst:
```diff
@@ -455,18 +455,21 @@ should be done:
 
-  * Packages that are bundled into the web client should generally use the
+  * Packages that are bundled into the web client must be listed under the ``dependencies`` field
+    of ``girder/web_client/src/package.json`` and should generally use the
```
Should this now just be done with the `npm add` command?

brianhelba replied:
There is no `npm add`, just an `npm install --save`. We don't want that, because there's no need to create a `node_modules` or `package-lock.json` within the `girder/web_client/src/` directory, which the command will always do. Since we don't need the functionality to keep those things in sync, hand-editing the file is going to be much easier than working with the statefulness of `npm install`.

zachmullen replied:
Makes sense, thanks.

brianhelba commented:
@girder/developers I'm having trouble reproducing the failing tests locally.

zachmullen commented:
The `girder build --dev` command fails silently (which is a separate bug).

<img width="408" alt="screen shot 2019-01-22 at 8 45 46 am" src="https://user-images.githubusercontent.com/555959/51539509-36493700-1e22-11e9-875e-a927d884d95e.png">

zachmullen commented on girder/web_client/grunt_tasks/webpack.config.js:
```diff
@@ -19,12 +19,9 @@
  * the DllPlugin for dynamic loading, each individual bundle has its own config options
  * that can extend these.
  */
-var path = require('path');
```
This line needs to be restored

brianhelba commented:
@zachmullen For some odd reason, [another build of the same code](https://circleci.com/gh/girder/girder/16752) (missing the `require('path')`), had the build step succeed without that error.

I've just fixed that particular issue, which was introduced when I fixed an issue with my original definition for `loaderPaths`.

At this point, even with the fix, the build is still failing. I'll continue to investigate, but again, any help is appreciated.

brianhelba commented:
I've managed to fix an issue affecting test loading. Without this fix, the build works for normal usage, but not most tests.

However, the tests still seem to be broken on the "integrationTests" builds. I'm investigating now to see what's different about that environment to cause the web client build to be different (`girder_lib.min.js` is a smaller size when built in that environment).

brianhelba commented:
I think the problem is that the machine executors on CircleCI are using NodeJS 6. Upgrading to Node 10 seems to fix the build.

brianhelba commented:
@girder/developers Tests are now all passing.
